CesiumGltf/include/CesiumGltf/FeatureIDTextureView.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include "FeatureIDTexture.h"
+#include "Image.h"
+#include "ImageCesium.h"
+#include "Model.h"
+#include "Texture.h"
+#include "TextureAccessor.h"
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <string>
+
+namespace CesiumGltf {
+
+enum class FeatureIDTextureViewStatus {
+  Valid,
+  InvalidUninitialized,
+  InvalidTextureIndex,
+  InvalidImageIndex,
+  InvalidChannel
+};
+
+class FeatureIDTextureView {
+public:
+  FeatureIDTextureView() noexcept
+      : _pImage(nullptr),
+        _channel(0),
+        _textureCoordinateIndex(-1),
+        _featureTableName(),
+        _status(FeatureIDTextureViewStatus::InvalidUninitialized) {}
+
+  FeatureIDTextureView(
+      const Model& model,
+      const FeatureIDTexture& featureIDTexture) noexcept
+      : _pImage(nullptr),
+        _channel(0),
+        _textureCoordinateIndex(-1),
+        _featureTableName(featureIDTexture.featureTable),
+        _status(FeatureIDTextureViewStatus::InvalidUninitialized) {
+
+    this->_textureCoordinateIndex =
+        featureIDTexture.featureIds.texture.texCoord;
+
+    int32_t textureIndex = featureIDTexture.featureIds.texture.index;
+    if (textureIndex < 0 || textureIndex >= model.textures.size()) {
+      this->_status = FeatureIDTextureViewStatus::InvalidTextureIndex;
+      return;
+    }
+
+    const Texture& texture = model.textures[textureIndex];
+
+    // Ignore sampler, we will always use nearest pixel sampling.
+    if (texture.source < 0 || texture.source >= model.images.size()) {
+      this->_status = FeatureIDTextureViewStatus::InvalidImageIndex;
+      return;
+    }
+
+    this->_pImage = &model.images[texture.source].cesium;
+
+    // TODO: is this right?
```

Comment:
That's an interesting one. I also stumbled over this, c.f. https://github.com/CesiumGS/3d-tiles-samples/pull/40#issuecomment-1019315404 . Actually, the spec does not say much about this, so I looked into the glTF spec and ... this does not say much about this either: It talks about red, green, blue and alpha channels, but does not _explicitly_ say anything about how many channels there "are" (or might be), or the order of these channels (particularly for the case that there are fewer than 4 channels). 

Vaguely related statements from https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#metallic-roughness-material might be...

> The base color texture MUST contain 8-bit values encoded with the sRGB opto-electronic transfer function ..

(even though that does not say, for example, whether an alpha channel is present), and

> For example, assume an 8-bit RGBA value of [64, 124, 231, 255] is sampled from baseColorTexture  ...

which indicates that RGBA may be stored there, in this order, but that's only an example, and not sufficient for answering the underlying question.

Surprisingly, the part at https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_texture_basisu/README.md#ktx-v2-images-with-basis-universal-supercompression seems to be more specific in terms of matching the different textures with their types, which channels are present, and the order of these channels. The answer to the question from the code...

>     // For instance, if the image has two channels, will those be referred to
>     // as r and g or r and a? 

might be that it is "r and g" based on that, but that's just a gut feeling for now.

(It might even be that this question is worth being "escalated" it to an issue in the glTF repo...)